app/src/main/java/com/islaharper/dawnofandroid/presentation/screens/welcome/WelcomeScreen.kt

```diff
@@ -1,12 +1,176 @@
+import com.google.accompanist.pager.HorizontalPagerIndicator
+import com.google.accompanist.pager.PagerState
+import com.google.accompanist.pager.rememberPagerState
+import com.islaharper.dawnofandroid.R
+import com.islaharper.dawnofandroid.domain.model.OnBoardingPage
+import com.islaharper.dawnofandroid.navigation.Screen
+import com.islaharper.dawnofandroid.ui.theme.AndroidGreen
+import com.islaharper.dawnofandroid.ui.theme.Neutral90
+import com.islaharper.dawnofandroid.ui.theme.PADDING_SMALL
+import com.islaharper.dawnofandroid.ui.theme.PADDING_XLARGE
+import com.islaharper.dawnofandroid.util.Constants.ONBOARDING_LAST_PAGE
+import com.islaharper.dawnofandroid.util.Constants.ONBOARDING_PAGE_COUNT
 
+@OptIn(ExperimentalPagerApi::class)
 @Composable
 fun WelcomeScreen(
     navHostController: NavHostController,
+    welcomeViewModel: WelcomeViewModel = hiltViewModel(),
 ) {
-    Text(text = "Welcome Placeholder")
+    val pages = listOf(
+        OnBoardingPage.FirstPage,
+        OnBoardingPage.SecondPage,
+        OnBoardingPage.ThirdPage,
+    )
+    val pagerState = rememberPagerState()
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(color = MaterialTheme.colorScheme.background),
+    ) {
+        HorizontalPager(
+            modifier = Modifier.weight(10f),
+            count = ONBOARDING_PAGE_COUNT,
+            state = pagerState,
+            verticalAlignment = Alignment.Top,
+        ) { position ->
+            PagerScreen(page = pages[position])
+        }
+        HorizontalPagerIndicator(
+            modifier = Modifier.weight(1f).align(Alignment.CenterHorizontally),
+            pagerState = pagerState,
+            activeColor = AndroidGreen,
+            inactiveColor = Neutral90,
+            indicatorWidth = 12.dp,
+            spacing = 8.dp,
+        )
+        FinishButton(
+            modifier = Modifier.weight(1f).padding(horizontal = PADDING_XLARGE),
+            pagerState = pagerState,
+        ) {
+            welcomeViewModel.saveOnBoardingState(completed = true)
+            navHostController.popBackStack()
+            navHostController.navigate(route = Screen.Login.route)
+        }
+    }
+}
+
+@Composable
+fun PagerScreen(page: OnBoardingPage) {
+    Column(
+        modifier = Modifier.fillMaxWidth(),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Center,
+    ) {
+        Image(
+            modifier = Modifier
+                .fillMaxWidth(0.5f)
+                .fillMaxHeight(0.7f),
+            painter = painterResource(id = page.image),
+            contentDescription = stringResource(R.string.onboarding_page),
+        )
+        Text(
+            modifier = Modifier
+                .fillMaxWidth(),
+            text = page.title,
+            color = MaterialTheme.colorScheme.onBackground,
+            fontSize = MaterialTheme.typography.displayMedium.fontSize,
```

Comment:
Can we use `fontStyle` here instead of unpacking the theme?

Reply:
Change made in latest commit.